assets/src/edit-story/components/inspector/inspectorProvider.js

```diff
@@ -70,6 +75,28 @@ function InspectorProvider({ children }) {
   const [inspectorContentHeight, setInspectorContentHeight] = useState(null);
   const inspectorContentRef = useRef();
   const tabRef = useRef(tab);
+  const firstPublishAttemptRef = useRef(false);
+
+  useEffect(() => {
+    // If a user wants to review their checklist before publishing
+    // a story that has high priority checklist items in it
+    // we need to go to the checklist tab and focus it.
+    // Because of how context is wrapped around the header
+    // we need to do this by watching the isChecklistReviewRequested value
+    // from the prepublishChecklistProvider.
+    if (isChecklistReviewRequested && !firstPublishAttemptRef.current) {
+      setTab(PREPUBLISH);
+      // Focus prepublish which is the last item in the panel title list
+      inspectorRef.current?.firstChild?.lastChild?.focus();
+      firstPublishAttemptRef.current = isChecklistReviewRequested;
+    }
+    // If a published story that gets reverted to a draft and
+    // it has high priority checklist items in it we should update
+    // this ref so that the checklist panel focuses again.
+    else if (!isChecklistReviewRequested && firstPublishAttemptRef.current) {
+      firstPublishAttemptRef.current = false;
+    }
+  }, [isChecklistReviewRequested]);
```

Comment:
💭 Could this use `setTab` to switch the inspector over to pre-publish?

Reply:
:( I tried `setTab` first but alas, no bueno. 

setting the tab just says that 'hey the checklist tab is active' - it doesn't focus anything. Plus, hijacking the focus every time the tab is open seems counterintuitive to the tab structure. 

And yeah, since the inspector ref isn't visible to the prechecklist provider we can't set it there. 

I figured this was an ok compromise because we already have the inspectorRef here and it keeps functionality for this set of panels in the same spot? 

There's a chance that the prepublish checklist gets moved out of this panel in the next iteration, so fingers crossed this is a temporary thing.

Reply:
oh that's not a bad idea. publish status is coming from the story anyway right? just inspect the current story and previous story in a register and anytime the status flips from something else to published fire an event? something like:
```
STORY_EVENTS.onPublish
```

Reply:
this focus happens _before_ the story gets published though.  

maybe i'm missing something? It's definitely one of those days 🤕